API Management SDK for version 2024-05-01

solankisamir opened the pull request:
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

solankisamir committed:
initial sdk and gateway resource

solankisamir committed:
initial sdk and gateway resource

solankisamir committed:
rename backend and certificate and ran export api

solankisamir committed:
rename backend and certificate and ran export api

nickliu-msft committed:
[Storage] Integrate decryption for CSE v2.1 (#46064)

* Initial commit

* Commit refactor

* removed obselete

* Added Changelog

* Fixed some LiveOnly tests

* Changelog added for Blobs and Queues

* Reverting changelog for Common and Queues

JimSuplizio committed:
Give CODEOWNERS file an owner in CODEOWNERS (#46205)

* Give CODEOWNERS file an owner in CODEOWNERS

* Remove a blank line, that already existed, because it'll make Jesse both warm and fuzzy

ShivangiReja committed:
[Search] Preview Features for API Version 2024-09-01-preview (#46107)

[Search] Preview Features for API Version 2024-09-01-preview

Co-authored-by: ShivangiReja <shivangi.reja@microsoft.com>

christothes committed:
Fix CM provisioning (#46206)

azure-sdk committed:
Increment package version after release of Azure.Search.Documents (#46208)

ahsonkhan committed:
Updated the request header sent to the OIDC endpoint so it doesn't result in a redirect response when an invalid system access token is provided. (#46135)

* Updated the request header sent to the OIDC endpoint in so it doesn't
result in a redirect response when an invalid system access token is
provided.

* Fix typo in header name and update test name.

* Update TSG wording to match error, and fix assert.

* Change the test back to returning void.

* Update wording in TSG to reset CI.

solankisamir committed:
resolved all apicompat issues

solankisamir committed:
resolved all apicompat issues

solankisamir committed:
Fix for Operation Status API

solankisamir committed:
Fix for Operation Status API

azure-sdk commented:
**API change check**

APIView has identified API level changes in this PR and created following API reviews.

[Azure.ResourceManager.ApiManagement](https://apiview.dev/Assemblies/Review/441201e06d7e48a895102a33026828c0?revisionId=fe446daf43bd4d01ad9a6af1a6b9c642)

solankisamir committed:
rename ApiManagementGatewayConfig to GatewayConfig

solankisamir committed:
rename ApiManagementGatewayConfig to GatewayConfig

welovej committed:
Enable-WriteCore-feature-for-desktopvirtualization (#46146)

welovej committed:
Enable-WriteCore-feature-for-devcenter (#46147)

welovej committed:
Enable-WriteCore-feature-for-devspaces (#46148)

azure-sdk committed:
Update AutoRest C# version to 3.0.0-beta.20240924.2 (#46231)

* Update Generator Version 3.0.0-beta.20240924.2

* Update SDK codes co_de_1

melina5656 committed:
Enable WriteCore feature for deviceprovisioningservices (#46150)

melina5656 committed:
Enable WriteCore feature for deviceregistry (#46152)

welovej committed:
Enable-WriteCore-feature-for-dnsresolver (#46153)

melina5656 committed:
Enable WriteCore feature for edgezones (#46159)

melina5656 committed:
Enable WriteCore feature for elastic (#46160)

melina5656 committed:
Enable WriteCore feature for elasticsan (#46161)

welovej committed:
Enable-WriteCore-feature-for-extendedlocation (#46162)

welovej committed:
Enable-WriteCore-feature-for-digitaltwins (#46149)

welovej committed:
Enable-WriteCore-feature-for-fluidrelay (#46164)

welovej committed:
Enable-WriteCore-feature-for-frontdoor (#46165)

welovej committed:
Enable-WriteCore-feature-for-managedservices (#46190)

melina5656 committed:
Enable WriteCore feature for managednetwork (#46189)

welovej committed:
Enable-WriteCore-feature-for-machinelearningservices (#46188)

melina5656 committed:
Enable WriteCore feature for maintenance (#46187)

welovej committed:
Enable-WriteCore-feature-for-machinelearningcompute (#46186)

welovej committed:
Enable-WriteCore-feature-for-loadtestservice (#46183)

melina5656 committed:
Enable WriteCore feature for hybridaks (#46182)

welovej committed:
Enable-WriteCore-feature-for-kusto (#46181)

welovej committed:
Enable-WriteCore-feature-for-kubernetesconfiguration (#46179)

welovej committed:
Enable-WriteCore-feature-for-iot (#46176)

welovej committed:
Enable-WriteCore-feature-for-hybridkubernetes (#46173)

welovej committed:
Enable-WriteCore-feature-for-hybridconnectivity (#46172)

melina5656 committed:
Enable WriteCore feature for healthbot (#46171)

welovej committed:
Enable-WriteCore-feature-for-hdinsight (#46170)

welovej committed:
Enable-WriteCore-feature-for-guestconfiguration (#46168)

welovej committed:
Enable-WriteCore-feature-for-graphservices (#46167)

welovej committed:
Enable-WriteCore-feature-for-grafana (#46166)

melina5656 committed:
Enable WriteCore feature for deviceupdate (#46157)

welovej committed:
Enable-WriteCore-feature-for-dynatrace (#46156)

welovej committed:
Enable-WriteCore-feature-for-managementpartner (#46191)

welovej committed:
Enable-WriteCore-feature-for-maps (#46192)

welovej committed:
Enable-WriteCore-feature-for-hybridnetwork (#46175)

* Enable-WriteCore-feature-for-hybridnetwork

* Update assets.json

welovej committed:
Enable-WriteCore-feature-for-iotcentral (#46177)

welovej committed:
Enable-WriteCore-feature-for-keyvault (#46180)

melina5656 committed:
Enable WriteCore feature for hybridcompute (#46185)

welovej committed:
Enable-WriteCore-feature-for-networkanalytics (#46219)

welovej committed:
Enable-WriteCore-feature-for-nginx (#46222)

welovej committed:
Enable-WriteCore-feature-for-notificationhubs (#46223)

welovej committed:
Enable-WriteCore-feature-for-openenergyplatform (#46224)

welovej committed:
Enable-WriteCore-feature-for-oracle (#46226)

welovej committed:
Enable-WriteCore-feature-for-paloaltonetworks.ngfw (#46232)

welovej committed:
Enable-WriteCore-feature-for-playwrighttesting (#46233)

welovej committed:
Enable-WriteCore-feature-for-policyinsights (#46234)

welovej committed:
Enable-WriteCore-feature-for-postgresql (#46235)

welovej committed:
Enable-WriteCore-feature-for-powerbidedicated (#46236)

welovej committed:
Enable-WriteCore-feature-for-quantum (#46239)

welovej committed:
Enable-WriteCore-feature-for-qumulo (#46240)

welovej committed:
Enable-WriteCore-feature-for-providerhub (#46237)

welovej committed:
Enable-WriteCore-feature-for-marketplace (#46193)

welovej committed:
Enable-WriteCore-feature-for-marketplaceordering (#46194)

welovej committed:
Enable-WriteCore-feature-for-mediaservices (#46195)

welovej committed:
Enable-WriteCore-feature-for-migrationdiscoverysap (#46211)

welovej committed:
Enable-WriteCore-feature-for-mixedreality (#46212)

welovej committed:
Enable-WriteCore-feature-for-monitor (#46214)

welovej committed:
Enable-WriteCore-feature-for-netapp (#46215)

melina5656 committed:
Enable WriteCore feature for managednetworkfabric (#46217)

welovej committed:
Enable-WriteCore-feature-for-recoveryservices-siterecovery (#46245)

welovej committed:
Enable-WriteCore-feature-for-networkfunction (#46220)

welovej committed:
Enable-WriteCore-feature-for-relay (#46248)

welovej committed:
Enable-WriteCore-feature-for-reservations (#46249)

welovej committed:
Enable-WriteCore-feature-for-quota (#46241)

welovej committed:
Enable-WriteCore-feature-for-redis (#46246)

welovej committed:
Enable-WriteCore-feature-for-recoveryservices-datareplication (#46244)

ArcturusZhang committed:
Support azure primitive types (#45958)

* add test cases

* add more test cases

* clean using statements

* resolve comments

* revert some unexpected changes

* add some scripts to run the test cases

* resolve comments

welovej committed:
Enable-WriteCore-feature-for-resourcehealth (#46252)

welovej committed:
Enable-WriteCore-feature-for-resourcegraph (#46251)

welovej committed:
Enable-WriteCore-feature-for-resourceconnector (#46250)

azure-sdk committed:
Update Generator Version 3.0.0-beta.20240924.3 (#46253)

azure-sdk committed:
Update AutoRest C# version to 3.0.0-beta.20240925.3 (#46273)

* Update Generator Version 3.0.0-beta.20240925.3

* Update SDK codes he_ma_3

* Update SDK codes de_he_2

* Update SDK codes ad_co_0

* Update SDK codes ma_pu_4

scottaddie committed:
Remove stray asterisks from Azure OpenAI README heading (#46274)

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.RedisEnterprise (#46258)

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.Elastic (#46260)

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.ElasticSan (#46261)

nickliu-msft committed:
[Storage] CSE v2.1 Prepare for Release 2024-09-25 (#46275)

* update changelog

* updated csproj

* point to package reference

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.Resources (#46210)

azure-sdk committed:
Sync eng/common directory with azure-sdk-tools for PR 9035 (#46276)

* don't verify changelog for packages that aren't actively published anymore
---------

Co-authored-by: Scott Beddall <scbedd@microsoft.com>
Co-authored-by: Scott Beddall <45376673+scbedd@users.noreply.github.com>
Co-authored-by: Ben Broderick Phillips <ben@benbp.net>

azure-sdk committed:
Increment package version after release of Azure.Storage.Blobs (#46279)

nickliu-msft committed:
Revert Package reference after patch release (#46281)

azure-sdk committed:
Update azure-sdk-build-tools Repository Resource Refs in Yaml files (#46282)

NneOkeke committed:
Computeschedule .NET SDK (#45962)

azure-sdk committed:
Update AutoRest C# version to 3.0.0-beta.20240925.5 (#46309)

azure-sdk committed:
Increment version for webpubsub releases (#46300)

* Increment package version after release of Microsoft.Azure.WebPubSub.Common

* Increment package version after release of Microsoft.Azure.WebPubSub.AspNetCore

Y-Sindo committed:
Update WebPubSub.AspNetCore reference to WebPubSub.Commom (#46298)

ArthurMa1978 committed:
Fix cspell for ComputeSchedule (#46312)

jsquire committed:
[CODEOWNERS] Remove invalid owner (#46198)

The focus of these changes is to remove an account that is no longer associated with Microsoft or has had permissions revoked.

nickliu-msft committed:
[Storage] [DataMovement] [API-Change] Changed BlobsStorageResourceProvider.FromBlob() & BlobsStorageResourceProvider.FromContainer() to take a Uri over string (#46318)

* initial commit

* Fixed changelog

azure-sdk committed:
Update CodeownersLinter to 1.0.0-dev.20240926.2 (#46323)

Co-authored-by: James Suplizio <jasupliz@microsoft.com>

azure-sdk committed:
Docs onboarding: ValidatePackageForOnboarding2 -> PackageIsValidForDocsOnboarding (#46321)

Co-authored-by: Daniel Jurek <djurek@microsoft.com>

KrzysztofCwalina committed:
AZD-CM integration (#46106)

* added solution, and a placeholder test file

* added basic azd integration

* made principalId an azd parameter

* removed dependency on concrete config system

* removed accidentaly added file

* refactored

* started experimenting with storage APIs

* changed resource names to be == to CMID

* small PR feedback

* fixed build break

* merged

* removed unfinished work

* added cache and messaging

* use cache

* PR feedback

* cached SB Client

* updated api file

christothes committed:
BearerTokenAuthenticationPolicy handles CAE challenges (#46277)

pharring committed:
Use simpler delegate when constructing AzureEventSourceListener (#46202)

* Use simpler delegate when constructing AzureEventSourceListener

* Updated CHANGELOGs

azure-sdk committed:
Update AutoRest C# version to 3.0.0-beta.20240926.1 (#46331)

* Update Generator Version 3.0.0-beta.20240926.1

* Update SDK codes he_ma_3

* Update SDK codes de_hd_2

* Update SDK codes ad_co_0

* Update SDK codes pu_sp_5

* Update SDK codes ma_pu_4

* Update SDK codes co_da_1

* Update SDK codes sq_wo_6

amnguye committed:
DataMovement Stress Testing Part 1 (#46287)

* Added separate sln file for Storage Data Movement and start of stress testing for DM Blobs

* Fix to Stress Test README

* Fix path for .slnf file

* WIP

* WIP

* WIP

* WIP

* WIP

* WIP

* WIP - builds

* WIP - docker issues

* WIP - build issue in docker

* WIP - builds locally, not in Docker

* WIP - runs until hitting excuting yaml file

* WIP - helm won't commit run

* WIP - fixing upload scenario test

* WIP - helm still not pushing commit; added more options for scenarios

* WIP

* Cleanup

* WIP

* More cleanup and some PR Comments

* Part 2 PR comments

Commit:
[AzureMonidorDistro] sync vendored code (#46207)

* update resources.azure

* update shared

* changelog

* typo

jianingwang123 committed:
fix/45747 (#46339)

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.ComputeSchedule (#46335)

azure-sdk committed:
Increment package version after release of Azure.Communication.Email (#46329)

tg-msft committed:
Azure.Provisioning updates (#46317)

 Cleanup for preview
- Move Deployment to its own library
- Remove implicit default infrastructure
- Remove ProvisioningContext from Compile now that it's meant to be used only with Build
- Make sure everything out of preview has ResourceVersions
- Use `Provisioning` prefix consistently on happy path types
- Unpin CM

jalauzon-msft committed:
[Storage][DataMovement] Fix failing Live tests (#46325)

jalauzon-msft committed:
[Storage][DataMovement] Added support for disabling the transfer checkpointer (#46286)

christothes committed:
Port forward the 1.12.1 hotfix (#46346)

jalauzon-msft committed:
[Storage] Use try/finally for locks in `LocalTransferCheckpointer` (#46350)

jalauzon-msft committed:
[Storage] Fix Storage base package perf pipelines (#46351)

zhz0704 committed:
Fix for #46313 (#46360)

* Change ISO to lowercase in AddressCountryRegion.Serialization.cs

* Use CodeGenSerialization

* Remove unnecessary using

zackliu committed:
Use a better way to fix negotiation result (#46216)

* Use a better way to fix negotiation result

* fix a package reference

* Better refactor codes

* Add a fix for python

zhz0704 committed:
Update changelog for Azure.Maps.Search (#46361)

* Update changelog

* Fix typo

welovej committed:
Enable-WriteCore-feature-for-mobilenetwork (#46213)

welovej committed:
Enable-WriteCore-feature-for-redisenterprise (#46247)

welovej committed:
Enable-WriteCore-feature-for-dns (#46151)

welovej committed:
Enable-WriteCore-feature-for-resources (#46254)

welovej committed:
Enable-WriteCore-feature-for-search (#46255)

welovej committed:
Enable-WriteCore-feature-for-securitydevops (#46259)

welovej committed:
Enable-WriteCore-feature-for-selfhelp (#46262)

welovej committed:
Enable-WriteCore-feature-for-servicefabric (#46264)

welovej committed:
Enable-WriteCore-feature-for-servicenetworking (#46266)

welovej committed:
Enable-WriteCore-feature-for-signalr (#46267)

welovej committed:
Enable-WriteCore-feature-for-sqlmanagement (#46270)

welovej committed:
Enable-WriteCore-feature-for-sqlvirtualmachine (#46290)

welovej committed:
Enable-WriteCore-feature-for-storageactions (#46291)

welovej committed:
Enable-WriteCore-feature-for-storagemover (#46292)

welovej committed:
Enable-WriteCore-feature-for-storagepool (#46293)

welovej committed:
Enable-WriteCore-feature-for-subscription (#46296)

welovej committed:
Enable-WriteCore-feature-for-streamanalytics (#46295)

welovej committed:
Enable-WriteCore-feature-for-storagesync (#46294)

welovej committed:
Enable-WriteCore-feature-for-workloads (#46307)

welovej committed:
Enable-WriteCore-feature-for-workloadmonitor (#46306)

welovej committed:
Enable-WriteCore-feature-for-voiceservices (#46304)

welovej committed:
Enable-WriteCore-feature-for-trustedsigning (#46303)

welovej committed:
Enable-WriteCore-feature-for-synapse (#46299)

welovej committed:
Enable-WriteCore-feature-for-support (#46297)

zackliu committed:
Update release notes (#46362)

azure-sdk committed:
Increment package version after release of Azure.Maps.Search (#46363)

zhz0704 committed:
Update TimeZone changelog to unreleased (#46367)

zhz0704 committed:
Implement azure maps weather (#45895)

* Finish generating code

* Finish client

* Finish client

* Finish client

* Change coordinates to GeoPosition

* Add Models namespace

* Change some classes to internal

* Generate Azure.Maps.Weather.netstandard2.0.cs

* Update changelog.md

* Add WeatherLanguage.cs

* Minor fixes

* Finish adding options

* Fix options summary

* Add base options

* Finish tests and samples

* Add properties and fix WeatherClientLiveTestsBase.cs

* Finish testing and fix errors

* Update readme

* Minor fixes

* Update cspell.json

* Rename samples md

* Fix typo

* Resolve comments

* Update assets.json

* Fix ci errors

* Update assets.json

* Update sdk/maps/Azure.Maps.Weather/src/Models/Options/GetDailyIndicesOptions.cs

Co-authored-by: Charlie Chen <yuchungchen@microsoft.com>

* Resolve comments

* Update imports

* Fix ci errors

* Resolve comments

* Update assets.json

* Remove CodeGenModel and unnecessary usings

* Update samples

* Remove Azure.Maps.Weather from ci.yml

* Rename Druation to DruationInDays in GetAirQualityDailyForecastsOptions.cs

---------

Co-authored-by: Charlie Chen <yuchungchen@microsoft.com>

alrod committed:
[BUG] GetProcessorCount for webjobs service bus extension does not override processor count for flex consumption (Fixes #45970) (#46197)

* [BUG] GetProcessorCount for webjobs service bus extension does not override processor count for flex consumption #45970 (Fixes #45970)

* Adding bug description

* Adding UtilityTest

* Chasngelog

azure-sdk committed:
Increment package version after release of Microsoft.Azure.WebJobs.Extensions.WebPubSubForSocketIO (#46366)

nickliu-msft committed:
initial commit (#46375)

m-redding committed:
Update System.Memory.Data (#46134)

* link fixes

* Revert "link fixes"

This reverts commit 61d9eaaf21843daf3ec8f7997b2f1be2f842c4d9.

* update

* AOT updates

* SCM test fix

* update API

* Another storage fix

* fixes for communication / openai

* update snippets

* fix

* fixes

trrwilson committed:
Azure OpenAI: 2.0.0 stable release (#46378)

* initial cross-apply of 2.0.0 changes

* minor: changelog update

* correct Packages.Data.props

* export-api

* correct cross-repo assets.json change

* prepare-release and assets re-fix

* temporary exclusion for search projectref

* exclusion via AOAI folder rather than search

* exclusion from search test project

* move exclusion to proper itemgroup

* circumvent binarydata tostring issue on mac

christothes committed:
Log troubleshooting headers for AzurePipelinesCredential (#46376)

Commit:
Update test pipelines to default to federated authentication (#46326)

* Update tests pipelines to default to federated authentication

* rm explicit UseFederatedAuth set

nick863 committed:
Add open telemetry to Azure.AI.Inferencing (#45751)

* Add open telemetry to Azure AI Inference

ShivangiReja committed:
[Search] Fix nightly test failures (#46374)

Co-authored-by: ShivangiReja <shivangi.reja@microsoft.com>

tg-msft committed:
Azure.Provisioning: Quick Role fixes (#46352)

Azure.Provisioning: Quick Role fixes
- Rename AssignRole to CreateRoleAssignment
- Add optional suffix to CreateRoleAssignment
- Make ResourceName settable

trrwilson committed:
Azure OpenAI: 2.1.0-beta.1, including /realtime preview support (#46390)

* updates for 2.1.0-beta.1

* include export-api output

* link verification placation

* include restored assets.json for preview swap

* assets.json update

tg-msft committed:
Azure.Provisioning: Add name requirements for Postgres flexible server (#46396)

jsquire committed:
[Service Bus] Enhance logging for session timeouts (#46388)

* [Service Bus] Enhance logging for session timeouts

The focus of these changes is to enhance the logs
emitted when acquiring a session using the `ServiceBusReceiver`
times out or is canceled.  As these are known and
expected exception conditions, they should be logged
as verbose information rather than an error.

Previously, these scenarios were special-cased for
processors, but receivers were treated as standard
errors.  Going forward, the processor and receiver
scenarios are handled consistently.

benbp committed:
Default to sub config file paths/svc connection for public + sovereign live tests (#45161)

azure-sdk committed:
Update New-TestResources.ps1 to add quotes around shell envvar (#46377)

Co-authored-by: swathipil <swathip@microsoft.com>

annelo-msft committed:
fix for 46109 (#46401)

nisha-bhatia committed:
Update CODEOWNERS (#46403)

SpencerOfwiti committed:
Update Nginx SDK to API version 2024-06-01-preview (#46368)

* Update Nginx SDK to API version 2024-06-01-preview

* Update generated files

* Update assets json file

* Update API

* Update session recordings

* Function renaming

* Update recordings

drishtipeshwani09 committed:
Microsoft.ContainerInstance: 2024-05-01-preview api .NET SDK (#46343)

* Genrated the api and models based on latest SDK

* Fix api compat issue

* regen

* Update ContainerGroupData.cs

* Renamed OperatingSystemType to ContainerGroupOsType

* Added tests for container group profile resource

* Created a seperate file for cg profile collection tests

* Added cg profile operation tests

* Added test to get a specified cg profile revision

* Changed RecordedTest to Record for new tests added

* Revert "Changed RecordedTest to Record for new tests added"

This reverts commit 6baad05d11b1f3616c7878386afa7283e28e3bb4.

* Modifed the container group profile test base

* Updated the revision tests and assets tag

* Updated the cg profile list and revision tests

---------

Co-authored-by: Minghao Chen <30464227+HarveyLink@users.noreply.github.com>

kinelski committed:
[DocumentIntelligence] Renaming and fixing the behavior of TrainingOperation (#46399)

annelo-msft committed:
Preparation for release of Azure.Core 1.44.0 (#46411)

* Preparation for release of Azure.Core 1.44.0

* update CHANGELOG

annelo-msft committed:
Preparation for release of System.ClientModel 1.2.0 (#46410)

* System.ClientModel 1.2.0

* update CHANGELOG

nickliu-msft committed:
[Storage] [Blobs] [DataMovement] [Tests] End to End Tests for StartUploadDirectoryAsync & StartDownloadToDirectoryAsync (#46409)

* Initial commit

* Recorded tests

* Moved test over to DataMovement.Blobs

* Recorded in DataMovement.Blobs

* Fixed minor error with CreateBlobDirectoryTree

* Remove Azure.Storage.DataMovement recording

annelo-msft committed:
Preparation for release of Azure.Core.Experimental 0.1.0-preview.36 (#46412)

* Preparation for release of Azure.Core.Experimental 0.1.0-preview.36

* update CHANGELOG

amnguye committed:
[Storage] [DataMovement] [Blobs] Stress Scenarios (#46400)

* Added separate sln file for Storage Data Movement and start of stress testing for DM Blobs

* Fix to Stress Test README

* Fix path for .slnf file

* WIP

* WIP

* WIP

* WIP

* WIP

* WIP

* WIP - builds

* WIP - docker issues

* WIP - build issue in docker

* WIP - builds locally, not in Docker

* WIP - runs until hitting excuting yaml file

* WIP - helm won't commit run

* WIP - fixing upload scenario test

* WIP - helm still not pushing commit; added more options for scenarios

* WIP

* Cleanup

* WIP

* More cleanup and some PR Comments

* Part 2 PR comments

* WIP

* WIP

* WIP - Created all scenario types and implementation

* Cleanup

* Update script to include all new scenarios

* Cleanup

* PR Comments: Updated Create Blob Helper methods; removed file close; short hand

* PR comments: Removed unnecessary init transfermanager; reapplied options

azure-sdk committed:
Sync eng/common directory with azure-sdk-tools for PR 9092 (#46413)

* Export the subscription data from the service connection

* Update deploy-test-resources.yml

---------

Co-authored-by: Wes Haggard <Wes.Haggard@microsoft.com>
Co-authored-by: Wes Haggard <weshaggard@users.noreply.github.com>

tg-msft committed:
Azure.Provisioning: rename `ResourceName` to `IdentifierName` (#46415)

* Azure.Provisioning: rename ResourceName to IdentifierName
* Validate identifier names

kinelski committed:
[DocumentIntelligence] Reenabling ignored test after service fix (#46416)

KrzysztofCwalina committed:
Added support for optional KeyVault to CloudMachine (#46418)

* messaging

* wrapped keyvault

* removed test framework form the solution

* keyvault works

* removed unfinished send

* updated api file

azure-sdk committed:
Increment version for core releases (#46420)

* Increment package version after release of System.ClientModel

* Increment package version after release of Azure.Core.Experimental

* Increment package version after release of Azure.Core

annelo-msft committed:
Update Core library package versions for Az.Core 1.44.0 and SCM 1.2.0 releases. (#46421)

KrzysztofCwalina committed:
fixing build (#46423)

* fixing build

* generated api file

donnie-msft committed:
Fix minor typo in ImageAnalysisOptions.cs (#45339)

Just noticed this tiny typo while playing with your SDK.

dinazavyr committed:
Split public and int tests and fix public tests (#46408)

ShivangiReja committed:
[Search] Update Samples as per Azure OpenAI GA Version (#46397)

* [Search] Update Samples as per OpenAI GA Version

* Update ApiKeyCredential -> AzureKeyCredential

---------

Co-authored-by: ShivangiReja <shivangi.reja@microsoft.com>

tg-msft committed:
Azure.Provisioning release prep (#46427)

Azure.Provisioning release prep
* Temporarily disable name validation test (as that's also temporarily disabled).
* Add --prerelease everywhere
* Add whitespace changes from the release script and instructions to install the previous GA

jsquire committed:
[Extensions] October 2024 Release Prep (#46422)

The focus of these changes is to prepare the `Microsoft.Extensions.Azure` package
for its October 2024 release.

MatthewSteeples committed:
Dispose of streams that are created (#44510)

amnguye committed:
Updated Changelog (#46431)

jalauzon-msft committed:
[Storage][DataMovement] Add basic perf tests for DataMovement Blobs (#46424)

azure-sdk committed:
Need to add environment to subscription configuration (#46432)

Co-authored-by: Wes Haggard <Wes.Haggard@microsoft.com>

azure-sdk committed:
Increment package version after release of Microsoft.Extensions.Azure (#46433)

jsquire committed:
[EngSys] Bump central Extenstions version (#46435)

The focus of these changes is to bump the central version for the
`Microsoft.Extensions.Azure` package so that the improvements are picked up
by this month's Function extension package releases.

tg-msft committed:
Azure.Provisioning: Move name validation to Infrastructure and expose for Aspire (#46437)

Azure.Provisioning: Move name validation to Infrastructure and expose for Aspire

azure-sdk committed:
Increment version for provisioning releases (#46440)

* Increment package version after release of Azure.Provisioning

* Increment package version after release of Azure.Provisioning.SignalR

* Increment package version after release of Azure.Provisioning.AppConfiguration

* Increment package version after release of Azure.Provisioning.CosmosDB

* Increment package version after release of Azure.Provisioning.Search

* Increment package version after release of Azure.Provisioning.Redis

* Increment package version after release of Azure.Provisioning.Sql

* Increment package version after release of Azure.Provisioning.PostgreSql

* Increment package version after release of Azure.Provisioning.CognitiveServices

* Increment package version after release of Azure.Provisioning.KeyVault

* Increment package version after release of Azure.Provisioning.Storage

* Increment package version after release of Azure.Provisioning.WebPubSub

* Increment package version after release of Azure.Provisioning.ServiceBus

* Increment package version after release of Azure.Provisioning.EventHubs

azure-sdk committed:
Increment version for provisioning releases (#46441)

* Increment package version after release of Azure.Provisioning.ApplicationInsights

* Increment package version after release of Azure.Provisioning.Communication

* Increment package version after release of Azure.Provisioning.KubernetesConfiguration

* Increment package version after release of Azure.Provisioning.Deployment

* Increment package version after release of Azure.Provisioning.AppService

* Increment package version after release of Azure.Provisioning.ContainerRegistry

* Increment package version after release of Azure.Provisioning.ContainerService

* Increment package version after release of Azure.Provisioning.EventGrid

* Increment package version after release of Azure.Provisioning.OperationalInsights

azure-sdk committed:
Increment version for provisioning releases (#46442)

* Increment package version after release of Azure.Provisioning.ApplicationInsights

* Increment package version after release of Azure.Provisioning.Communication

* Increment package version after release of Azure.Provisioning.KubernetesConfiguration

* Increment package version after release of Azure.Provisioning.Deployment

* Increment package version after release of Azure.Provisioning.AppService

* Increment package version after release of Azure.Provisioning.ContainerRegistry

* Increment package version after release of Azure.Provisioning.ContainerService

* Increment package version after release of Azure.Provisioning.EventGrid

* Increment package version after release of Azure.Provisioning.OperationalInsights

* Increment package version after release of Azure.Provisioning.Kubernetes

* Increment package version after release of Azure.Provisioning.AppContainers

nickliu-msft committed:
[Storage] [DataMovement] [API-Change] Changed ShareFilesStorageResourceProvider.FromFile/Directory to take a Uri over string (#46446)

* Initial commit

* Export API

SpencerOfwiti committed:
Update Nginx changelog to support API version 2024-06-01-preview (#46447)

* Update Nginx changelog to support API version 2024-06-01-preview

* Refactor

jsquire committed:
[Service Bus] October 2024 Release Prep (#46438)

The focus of these changes is to prepare the Service Bus package for the October
2024 release.

seanmcc-msft committed:
Storage STG 96 (#45604)

nielsheeren committed:
Fix missing HybridSearch in copy method of SearchOptions (#46425)

Fix missing HybridSearch in copy SearchOptions

azure-sdk committed:
Update AutoRest C# version to 3.0.0-beta.20241007.1 (#46457)

* Update Generator Version 3.0.0-beta.20241007.1

* Update SDK codes he_ma_3

* Update SDK codes de_he_2

* Update SDK codes ma_pu_4

* Update SDK codes sq_wo_6

* Update SDK codes co_de_1

* export api

* fix tests

---------

Co-authored-by: Arcturus Zhang <dapzhang@microsoft.com>
Co-authored-by: Minghao Chen <30464227+HarveyLink@users.noreply.github.com>

Sid200026 committed:
Onboard Azure.Developer.MicrosoftPlaywrightTesting sdk (#45044)

* feat(): onboard Azure.Developer.MicrosoftPlaywrightTesting sdk

* chore(): add support for github summary

* docs(): sample, CHANGELOG and README

* docs(): content of samples

* chore():: modify samples as per azure sdk guidelines

* refactor(): unit tests as per azure sdk guidelines

* chore(): use autorest to generate api clients

* fix(): json parsing of error messages

* chore(): add live tests

* fix(): run id handling across scalable and reporting

* Adressing comments

* comment

* chore(): resolve documentation comments

* docs(): import c# code snippets from .cs files

* refactor(): update CODEOWNERS with latest labels

* chore(): add git commit based display name

* feat(): scalable run error handling

* refactor(): convert API client class to internal

* chore(): apiview review comments addressed for base sdk

* refactor(): rename tokenCredential to credential

* refactor(): rename PlaywrightServiceSettings to PlaywrightServiceOptions

* refactor(): use serviceAuth only

* refactor(): remove defaultAuth references

* refactor(): convert serviceOs class to internal

* refactor(): convert public fields into properties

* chore(): add cancellation token in async methods

* refactor(): use default for null cancellation tokens

* refactor(): added users in CODEOWNERS

* fix(): live test resource json

* docs(): add missing sections in README files

* refactor(): rename base package to Azure.Developer.MicrosoftPlaywrightTesting.TestLogger

* docs(): move authenticate client section after prerequisites

---------

Co-authored-by: Siddharth Singha Roy <ssingharoy@microsoft.com>
Co-authored-by: Vansh Vardhan Singh <vanshvsingh@gmail.com>

seanmcc-msft committed:
STG 96 beta changelogs (#46451)

* STG 96 beta changelogs

* PR comments

wps0 committed:
Azure.Monitor.Query: invalid date string format in MetricsClient (#45998)

azure-sdk committed:
Persist oidc env vars in deploy template (#46477)

Co-authored-by: Ben Broderick Phillips <bebroder@microsoft.com>

azure-sdk committed:
Increment version for storage releases (#46479)

* Increment package version after release of Azure.Storage.Common

* Increment package version after release of Azure.Storage.Files.Shares

* Increment package version after release of Azure.Storage.Blobs

* Increment package version after release of Azure.Storage.Queues

* Increment package version after release of Azure.Storage.Blobs.Batch

* Increment package version after release of Azure.Storage.Files.DataLake

* Increment package version after release of Azure.Storage.Blobs.ChangeFeed

azure-sdk committed:
Increment package version after release of Azure.Messaging.ServiceBus (#46480)

jsquire committed:
[Event Hubs] Convert tests to use identity auth (#46330)

* [Event Hubs] Convert tests to use identity auth

The focus of these changes is to convert the Event Hubs
live tests to use identity-based authorization by default.
Connection strings will continue to have test coverage and
appear in "Hello World" samples and in legacy samples for
the migration guide.

* Test fixes

* Fixing more typos

* Test debugging

* tuning test timing and renabling cancellation  check

* tuning test timing

* Extending test timing slightly

* Update sdk/eventhub/Azure.Messaging.EventHubs/tests/Snippets/Sample02_EventHubsClientsLiveTests.cs

Co-authored-by: Madalyn Redding Heaps <66138537+m-redding@users.noreply.github.com>

* Update sdk/eventhub/Azure.Messaging.EventHubs/tests/Snippets/Sample02_EventHubsClientsLiveTests.cs

Co-authored-by: Madalyn Redding Heaps <66138537+m-redding@users.noreply.github.com>

* Update sdk/eventhub/Azure.Messaging.EventHubs/tests/Snippets/Sample02_EventHubsClientsLiveTests.cs

Co-authored-by: Madalyn Redding Heaps <66138537+m-redding@users.noreply.github.com>

* Update sdk/eventhub/Azure.Messaging.EventHubs/tests/Snippets/Sample02_EventHubsClientsLiveTests.cs

Co-authored-by: Madalyn Redding Heaps <66138537+m-redding@users.noreply.github.com>

* Update sdk/eventhub/Azure.Messaging.EventHubs/tests/Snippets/Sample02_EventHubsClientsLiveTests.cs

Co-authored-by: Madalyn Redding Heaps <66138537+m-redding@users.noreply.github.com>

* Update sdk/eventhub/Azure.Messaging.EventHubs/tests/Snippets/Sample02_EventHubsClientsLiveTests.cs

Co-authored-by: Madalyn Redding Heaps <66138537+m-redding@users.noreply.github.com>

* Update sdk/eventhub/Azure.Messaging.EventHubs/tests/Snippets/Sample02_EventHubsClientsLiveTests.cs

Co-authored-by: Madalyn Redding Heaps <66138537+m-redding@users.noreply.github.com>

* Update sdk/eventhub/Azure.Messaging.EventHubs/tests/Snippets/Sample02_EventHubsClientsLiveTests.cs

Co-authored-by: Madalyn Redding Heaps <66138537+m-redding@users.noreply.github.com>

* Regenerating snippets

---------

Co-authored-by: Madalyn Redding Heaps <66138537+m-redding@users.noreply.github.com>

jsquire committed:
[EngSys] Update central Service Bus version (#46483)

The focus of these changes is to update the central version for the
`Azure.Messaging.ServiceBus` package to use the latest release.

azure-sdk committed:
Use equality function check for persist oidc token step (#46485)

Co-authored-by: Ben Broderick Phillips <bebroder@microsoft.com>

azure-sdk committed:
Sync eng/common directory with azure-sdk-tools for PR 9106 (#46482)

Co-authored-by: Scott Beddall <scbedd@microsoft.com>

m-redding committed:
Update version of System.Text.Json (#46508)

dubiety committed:
Update Maps codeowners (#46494)

jalauzon-msft committed:
[Storage][DataMovement] Fix perf test cleanup + other minor fixes (#46486)

jaschrep-msft committed:
TransferManager tests (#46449)

* test queue failure

* More tests

* tests

* more tests (failing)

* test fixes. some new testing infra.

move one property access inside a trycatch

* reenable check

* remove enum to depend on test-wide def

m-redding committed:
Prep for release (#46512)

JonathanCrd committed:
Remove heaths as codeowner and add myself (#46488)

weshaggard committed:
Only treat critical vulnerabilities as errors (#46515)

nickliu-msft committed:
[Storage] [DataMovement] Moved all blob related tests out of DataMovement.Tests to DataMovement.Blobs.Test (#46475)

* Initial commit

* Moved more test files

* More tests moved over

* Compile Include over just adding the files to DM.Blobs.Tests

solankisamir committed:
tests

welovej committed:
Enable-WriteCore-feature-for-networkcloud (#46469)

azure-sdk committed:
Update AutoRest C# version to 3.0.0-beta.20241009.1 (#46507)

* Update Generator Version 3.0.0-beta.20241009.1

* Update SDK codes he_ma_3

* Update SDK codes de_he_2

* Update SDK codes ad_co_0

* Update SDK codes pu_sp_5

* Update SDK codes co_de_1

* Update SDK codes ma_pu_4

* Update SDK codes sq_wo_6

* bump version of System.Text.Json

---------

Co-authored-by: Arcturus Zhang <dapzhang@microsoft.com>

frantran committed:
2024-05 Fleet .NET sdk generation (#46353)

* 2024-05 Fleet api .net sdk generation

* fix breaking change issue, add tests and changelog

* change changelog version

---------

Co-authored-by: frantran <ftran@microsoft.com>

Y-Sindo committed:
Support handling function return value from worker process (#46158)

Fix Azure/azure-functions-dotnet-worker#1496

The reason to add .NET 6 and .NET 8 as TFM:
The worker function return value is a Newtonsoft JObject if it's an object in the worker. Since .NET Core Frameworks, the default SignalR JSON protocol uses System.Text.Json library and can't serialize Newtonsoft JObject. Therefore, we need to wrap the JObject to a System.Text.Json serializable object with a customized JSON converter. The customized JSON converter first converts the Newtonsoft JObject to JSON with Newtonsoft library, then writes the raw JSON object with System.Text.Json writer. For .NET Standard 2.0, it's not possible to write a raw JSON with System.Text.Json writer, therefore, we need to add .NET 6 and .NET 8 as TFM to write the raw JSON.

archerzz committed:
fix(digitaltwins): bump System.Text.Json to 6.0.10 (#46523)

Co-authored-by: Mingzhe Huang (from Dev Box) <mingzhehuang@microsoft.com>

frantran committed:
add release date to Fleet .NET SDK latest version (#46524)

* add release date to latest version

* address version comment

* Update CHANGELOG.md

---------

Co-authored-by: frantran <ftran@microsoft.com>
Co-authored-by: Mingzhe Huang <archerzz@users.noreply.github.com>

azure-sdk committed:
Update AutoRest C# version to 3.0.0-beta.20241009.4 (#46526)

* Update Generator Version 3.0.0-beta.20241009.4

* Update SDK codes pu_sp_5

* Update SDK codes co_de_1

* regen

---------

Co-authored-by: Mingzhe Huang (from Dev Box) <mingzhehuang@microsoft.com>

azure-sdk committed:
Increment version for core releases (#46517)

* Increment package version after release of Azure.Core

* Increment package version after release of System.ClientModel

solankisamir committed:
Merge branch 'sasolank/apim-2024-05-01-sdk' of https://github.com/Azure/azure-sdk-for-net into sasolank/apim-2024-05-01-sdk

jsquire reviewed (changes requested):
This PR is not reviewable in the current format.  Please rebase onto `main` and ensure that only the changers that you're intending to make appear in the files.  You should not be including unrelated changes.
